include/xtypes/idl/parser.hpp

```diff
@@ -345,6 +909,204 @@ class Parser
+            const std::shared_ptr<peg::Ast> ast,
+            std::shared_ptr<SymbolScope> outer)
+    {
+        using namespace peg::udl;
+        std::string name;
+        //std::map<std::string, Member> member_list;
+        for (const auto& node : ast->nodes)
+        {
+            switch (node->original_tag)
+            {
+                case "IDENTIFIER"_:
+                {
+                    name = resolve_identifier(ast, node->token, outer, true);
+                    break;
+                }
+                case "BITFIELD"_:
+                    //bitfield(node, outer, member_list, type);
+                    // TODO:
+                    // + BITFIELD
+                    //     - BITFIELD_SPEC/0[POSITIVE_INT_CONST] (3)
+                    //     - IDENTIFIER (a)
+                    // + BITFIELD
+                    //     - BITFIELD_SPEC/0[POSITIVE_INT_CONST] (1)
+                    //     - IDENTIFIER (b)
+                    // - BITFIELD/0[POSITIVE_INT_CONST] (4)
+                    // + BITFIELD
+                    //     + BITFIELD_SPEC
+                    //         - POSITIVE_INT_CONST (10)
+                    //         + DESTINATION_TYPE/2[SIGNED_LONG_INT]
+                    //     - IDENTIFIER (c)
+                    break;
+            }
+        }
+
+        std::cout << "Found \"bitset " << name << "\" but bitsets aren't supported. Ignoring." << std::endl;
+        /* TODO
+        BitsetType bitset_type(name);
+        for (auto& member : member_list)
+        {
+            bitset_type.add_member(std::move(member.second));
+        }
+        // Replace
+        outer->bitsets[name] = DynamicType::Ptr(bitset_type);
+        */
+    }
+
+    void bitmask_dcl(
+            const std::shared_ptr<peg::Ast> ast,
+            std::shared_ptr<SymbolScope> outer)
+    {
+        using namespace peg::udl;
+        std::string name;
+        //std::map<std::string, Member> member_list;
+        for (const auto& node : ast->nodes)
+        {
+            switch (node->original_tag)
+            {
+                case "IDENTIFIER"_:
+                {
+                    name = resolve_identifier(ast, node->token, outer, true);
+                    break;
+                }
+                case "BIT_VALUE"_:
+                    //switch_body(node, outer, member_list, type);
+                    // TODO:
+                    //  - BIT_VALUE/0[IDENTIFIER] (flag0)
+                    //  - BIT_VALUE/0[IDENTIFIER] (flag1)
+                    //  + BIT_VALUE
+                    //      + ANNOTATION_APPL
+                    //          - SCOPED_NAME (position)
+                    //          - ANNOTATION_APPL_PARAMS/1[LITERAL] (5)
+                    //      - IDENTIFIER (flag5)
+                    //  - BIT_VALUE/0[IDENTIFIER] (flag6)
+                    break;
+            }
+        }
+
+        std::cout << "Found \"bitmask " << name << "\" but bitmasks aren't supported. Ignoring." << std::endl;
```

Comment:
Other option for the couts could be set a "verbose flag", that enable disable the output.

Reply:
The best solution here should be the creation of a "log" that the user can retrieve allowing filtering by categories (UNSUPPORTED, UNDECLARED, REDEFINITION, ALREADY_DECLARED...) and verbosity (DEBUG, INFO, WARNING, ERROR...).

Reply:
I like so much. Added task #6919